src/main/scala/firrtl/stage/phases/Checks.scala

```diff
@@ -0,0 +1,84 @@
+package firrtl.stage.phases
+
+import firrtl.AnnotationSeq
+import firrtl.annotations.Annotation
+import firrtl.options.{OptionsException, Phase, TargetDirAnnotation}
+import firrtl.stage._
+
+object Checks extends Phase {
+
+  /** Determine if an annotations are sane
+    *
+    * @param annos a sequence of [[annotation.Annotation]]
+    * @return true if all checks pass
+    */
+  def transform(annos: AnnotationSeq): AnnotationSeq = {
```

Comment:
I like this being a Transform

Reply:
Great! I'm glad this makes sense. And I assume you mean a `Phase` with a `transform` method?

FWIW, #945 grew out of the fact that I was constantly writing methods with the type signature of `AnnotationSeq => AnnotationSeq`. These weren't a `Stage`, but something else. Additionally, the `view` really has no business running checks. The common pattern seems to be that the type you're viewing an `AnnotationSeq` as is a `case class` with default values. Hence, the checks are something transform-like that live outside the `view`.